src/averageAges.js

```diff
@@ -14,14 +14,23 @@
  *
  * @return {number}
  */
+const MALE = 'f';
+const FEMALE = 'f';
+const ONE_CENTURY = 100;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const men = people.filter(person => person.sex === MALE && (century
+    ? Math.ceil(person.died / ONE_CENTURY) === century
+    : true));
+
+  const menTotalAge = men.reduce((acc, man) => {
+    const age = man.died - man.born;
+
+    return acc + age;
+  }, 0);
 
+  return menTotalAge / men.length;
```

Comment:
For the calculation of the men's average age, when no men are filtered (array length is 0), division by 0 will occur leading to NaN. Implement a guard to handle division when the men array is empty.